src/Indicator.vala

```diff
@@ -95,41 +91,59 @@ public class Power.Indicator : Wingpanel.Indicator {
 
         return popover_widget;
     }
+
     /* Smooth scrolling vertical support. Accumulate delta_y until threshold exceeded before actioning */
-    private double total_y_delta= 0;
```

Comment:
I think we should move line 96 till line 149 into a separate file. 
for me this clutters the indicator class.

Reply:
Yep!, Im working on that.

Reply:
Done!